### PR TITLE
responsive downloads table

### PR DESCRIPTION
**Pull request by sandikbarr**

### CODE REVIEW

File: src/app/downloads/table/downloads-table-container.component.ts [hunk 1]
```diff
@@ -57,9 +59,8 @@ export class DownloadsTableContainerComponent implements OnInit {
     this.store.dispatch(new ACTIONS.RouteEpisodePageAction({episodePage}));
   }
 
-  toggleExpandedReport() {
-    this.expanded = !this.expanded;
-    if (this.expanded) {
+  toggleExpandedReport(expanded) {
+    if (expanded) {
```
Comment (sandikbarr) [inline src/app/downloads/table/downloads-table-container.component.ts]: expanded state variable moved to child component

File: src/app/downloads/table/scrolling-table.component.css [hunk 2]
```diff
@@ -0,0 +1,50 @@
+:host {
+  display: block;
+  flex-grow: 1;
+  overflow-x: auto;
+}
+
```
Comment (sandikbarr) [inline src/app/downloads/table/scrolling-table.component.css]: This scrolling table layout didn't really change from the original. I just split it off to its own component. It worked with the new design well enough, and I think it's going away, maybe with #283?

File: src/app/ngrx/reducers/selectors/downloads-table.selectors.ts [hunk 3]
```diff
@@ -21,31 +23,39 @@ export const selectDownloadTablePodcastDownloads = createSelector(
...
+        if (totalForPeriod > podcastAllTimeDownloads.allTimeDownloads) {
+          podcastData.allTimeDownloads = totalForPeriod;
+        } else {
+          podcastData.allTimeDownloads = podcastAllTimeDownloads.allTimeDownloads;
         }
       }
     }
     return podcastData;
   }
 );
```
Comment (sandikbarr) [inline src/app/ngrx/reducers/selectors/downloads-table.selectors.ts]: The downloads array has a separate selector now. Building this any[][] with the formatted headers in the selector is better for performance. Previously these dates were formatted in the component and checked on change detection.

File: src/app/downloads/table/downloads-table-container.component.spec.ts [hunk 4]
```diff
@@ -4,14 +4,16 @@ import { RouterTestingModule } from '@angular/router/testing';
...
+import { SummaryTableComponent } from './summary-table.component';
 
-import { reducers } from '../ngrx/reducers';
-import { CHARTTYPE_EPISODES, EPISODE_PAGE_SIZE } from '../ngrx';
-import * as ACTIONS from '../ngrx/actions';
-import { routerParams, episodes, podcast, ep0Downloads, ep1Downloads, podDownloads } from '../../testing/downloads.fixtures';
+import { reducers } from '@app/ngrx/reducers';
+import { CHARTTYPE_EPISODES, EPISODE_PAGE_SIZE } from '@app/ngrx';
+import * as ACTIONS from '@app/ngrx/actions';
+import { routerParams, episodes, podcast, ep0Downloads, ep1Downloads, podDownloads } from '../../../testing/downloads.fixtures';
```
Comment (pstjean) [inline src/app/downloads/table/downloads-table-container.component.spec.ts]: Could use that slick new non-relative import here.

File: src/app/downloads/table/summary-table.component.ts [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+
+    <div class="row" *ngIf="podcastTableData">
+      <div>
+        <prx-checkbox *ngIf="showPodcastToggle; else podcastTitle"
+          small [checked]="podcastTableData.charted" [color]="podcastTableData.color"
+          (change)="toggleChartPodcast.emit($event)">
+          {{podcastTableData.title}}
+        </prx-checkbox>
+        <ng-template #podcastTitle><span class="title">{{podcastTableData.title}}</span></ng-template>
+        <span class="mobile title">{{podcastTableData.title}}</span>
```
Comment (pstjean) [inline src/app/downloads/table/summary-table.component.ts]: Should this be conditionally displayed as well?

File: src/app/shared/pipes/large-number.pipe.spec.ts [hunk 6]
```diff
@@ -12,9 +12,9 @@ describe('LargeNumberPipe', () => {
     expect(pipe.transform(value)).toEqual('1,000');
   });
 
-  it('returns zero for falsey values', () => {
-    expect(pipe.transform(undefined)).toEqual('0');
-    expect(pipe.transform(null)).toEqual('0');
-    expect(pipe.transform(Number(false))).toEqual('0');
+  it('returns zero for only for zero values', () => {
```
Comment (pstjean) [inline src/app/shared/pipes/large-number.pipe.spec.ts]: Extra “for” snuck in here.

File: src/tsconfig.spec.json [hunk 7]
```diff
@@ -3,6 +3,10 @@
   "compilerOptions": {
     "outDir": "../out-tsc/spec",
     "baseUrl": "./",
+    "paths": {
+      "@app/*": ["app/*"],
```
Comment (pstjean) [inline src/tsconfig.spec.json]: Handy! ✨

File: src/app/downloads/table/summary-table.component.ts (hunk 5 above)
Comment (sandikbarr) [reply src/app/downloads/table/summary-table.component.ts]: bleh, I don't love how this has 3 title fields 2 conditionally controlled by Angular, but the `.mobile.title` being displayed by CSS (along with the `.title:not(.mobile), prx-checkbox` being hidden.) It's confusing, and I didn't figure out a way around it. I added a comment about it and used a better class name. Here is that CSS: https://github.com/PRX/metrics.prx.org/pull/381/files#diff-3bc32d7b8aa81549c4ca5c265d15adf9R60

Comment (pstjean) [reply src/app/downloads/table/summary-table.component.ts]: Ah, the `:not` selector. That's what I was missing. Makes sense, if a bit awkward.

Comment (sandikbarr) [reply src/app/downloads/table/summary-table.component.ts]: definitely awkward, thought maybe the responsive table could be separate in the template... Changing flex-direction seemed to be fine except for hiding this functionality did come out a little messier.

**Pull request merged**